Predictors/rnn.py

```diff
@@ -0,0 +1,172 @@
+
+        self.init = tf.global_variables_initializer()
+
+    def fit(self, sentences, labels, sess=None):
+        """
+        sentences and labels should already be encoded
+        """
+
+        if sess is None:
+            sess = tf.get_default_session()
+
+        test_prop = 0.1
+        train_prop = 1.0 - test_prop
+        split = floor(train_prop * sentences.shape[0])
+
+        X_train = sentences[:split]
+        X_test = sentences[split:]
+        y_train = labels[:split]
+        y_test = labels[split:]
+
+        print("Training RNN Classifier")
+
+        sess.run(self.init)
+        for epoch in range(self.n_epochs):
+            try:
+                for iteration in range(sentences.shape[0] // self.batch_size):
+                    # pick a random batch
+                    batch_choices = np.random.choice(X_train.shape[0], self.batch_size, replace=True)
+                    X_batch = X_train[batch_choices]
+                    y_batch = y_train[batch_choices]
+                    # print("type(X_batch = {}, type(y_batch) = {}\n".format(type(X_batch), type(y_batch)))
+                    # print("X_batch.shape = {}, y_batch.shape = {}\n".format(X_batch.shape, y_batch.shape))
+                    # print("X_batch.dtype = {}, y_batch.dtype = {}\n".format(X_batch.dtype, y_batch.dtype))
+
+                    sess.run(self.training_op, feed_dict={self.X: X_batch, self.y: y_batch})
+
+                acc_train = self.accuracy.eval(feed_dict={self.X: X_batch, self.y: y_batch})
+                acc_test = self.accuracy.eval(feed_dict={self.X: X_test, self.y: y_test})
+                print(
+                    "{}/{}:\t\tTrain accuracy:\t{:.3f}\tTest accuracy:\t{:.3f}".format(epoch, self.n_epochs, acc_train,
+                                                                                       acc_test))
+
+            except KeyboardInterrupt:  # stop early with Control+C (SIGINT)
+                print("\nInterrupted by user at epoch {}/{}".format(epoch, self.n_epochs))
+                return self
+
+        return self
+
+    def predict_proba(self, X, sess=None):
+        """
+        probabillites for each class
+        """
+        if sess is None:
+            sess = tf.get_default_session()
+
+        self.init.run()
+        return sess.run(self.probs, feed_dict={self.X: X})
+
+    def predict(self, X, sess=None):
+        """
+        predict class
+        """
+        if sess is None:
+            sess = tf.get_default_session()
+
+        out = sess.run(self.output, feed_dict={self.X: X})
+        return out
+
+
+if __name__ == '__main__':
+    from data.numbered_authors import NumberAuthorsTransformer
+    from data.padded_sentences import PaddedSentenceTransformer
+    from data.import_data import import_data
+    from Predictors.sklearnclassifier import show_stats
+
+    tr, te = import_data()
+    # data_enc = WindowedSentenceTransformer(encoder_size=50)
```

Comment:
can we put WindowedSentenceTransformer and PaddedSentenceTransformer in an `if` statement? It's better than commenting and uncommenting this test code. It also allows much easier automation later.